### PR TITLE
Add server log automation

### PR DESCRIPTION
**Pull request by allisaurus**
*Description of changes:*

Previously, operators needed to RDP into the server in order to retrieve logs and metrics files generated during testing. This feature syncs test artifacts to a temporary S3 bucket while the server is deployed, and uses an AWS Lambda function to export them to another (external to this tool) S3 bucket when the server stack is destroyed, for longer term storage. 

## Testing

 - successfully deploys
 - manual tests done in Lambda  web console:
    - happy case
    - error case: required bucket not found
    - error case: `resources` event key missing or empty

--
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cdk/lambda/upload_test_artifacts/upload_test_artifacts.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        if export['Name'] == DEFAULT_DESTINATION_BUCKET_EXPORT_NAME:
+            destination_bucket = export['Value']
+            print(f'destination_bucket is: {destination_bucket}')
+ 
+    if destination_bucket == '' or source_bucket == '':
+        raise RuntimeError('Required bucket missing! No action will be taken.')
+ 
+    # upload bucket content with unique key
+    s3_client = boto3.client('s3')
+    list_response = s3_client.list_objects(Bucket=source_bucket)
```
Comment (lmbr-pip) [inline cdk/lambda/upload_test_artifacts/upload_test_artifacts.py:41]: nit: recommendation is to use https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/list_objects_v2.html over list_objects.

Also paginated API so you could miss results!

File: cdk/multiplayer_test_scaler/server_artifacts_automation.py [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        )
+
+        # can't use L2 events.Rule b/c it doesn't yet support
+        # SSM RunCommand targets. See: https://github.com/aws/aws-cdk/issues/7710
+        self._file_sync_rule = events.CfnRule(self, 'IntermitentUploadRule',
+            name=f'{RESOURCE_ID_COMMON_PREFIX}SyncArtifactsRule',
+            schedule_expression='rate(2 minutes)',
+            targets=[events.CfnRule.TargetProperty(
+                arn=doc_arn,
+                id='MpScalerFileSyncRule',
```
Comment (lmbr-pip) [inline cdk/multiplayer_test_scaler/server_artifacts_automation.py:59]: nit: MpsScaler

Comment (allisaurus) [reply cdk/multiplayer_test_scaler/server_artifacts_automation.py:59]: the headline title of the tool is "Multiplayer Test Scaler", so I think "MP Scaler" is the better abbreviation. I know historically we've referred to it as "Multiplayer sample scaler" to refer to the main test case (O3DE Multiplayer Sample), but the intention is to be more generically useful so I'd rather drop the "sample"

File: README.md [hunk 3]
```diff
@@ -103,6 +103,13 @@ This command will:
 - _config-file_: Path to the config file to use. If no config file is specified, the tool will search for an existing config file called `multiplayer_test_scaler_config.json` under the execution directory.
 - _platform_: Platform the project will be built for. Currently, only supports `Windows`.
 
+### Create and/or specify an Amazon S3 bucket for your test artifacts
+When deployed, Multiplayer Test Scaler will sync the contents of the `C:/o3de/user/log` and `C:/o3de/user/Metrics` directories on the server to a temporary S3 bucket in your account to ensure they are recoverable in case your server instance becomes unreachable during testing. When the server stack is destroyed, Multiplayer Test Scaler will attempt to upload the contents of this bucket to another (longer lived) S3 bucket in your account for long-term storage.
```
[the remark this answers is not in the document]
Comment (allisaurus) [reply README.md:107]: the longer lived bucket is not created by MP Scaler, it's looked up in the customer's account based on the Cfn export name "O3deMetricsUploadBucket" (bullets below explain how to create it or configure a diff name to query). The temp artifact bucket is available for the duration of the server stack lifetime, the lifetime of the long term storage bucket is not defined by MP Scaler but is assumed to be long lived so operators can hold on to the test artifacts for a while.

Will update this section to spell things out more clearly, as there are a couple diff things going on, one of which is the requirement that users create or define a bucket for long term storage.

File: README.md [hunk 4]
```diff
@@ -103,6 +103,15 @@ This command will:
...
 - _platform_: Platform the project will be built for. Currently, only supports `Windows`.
 
+### Create and/or specify an Amazon S3 bucket for exporting your test artifacts
+
+When deployed, Multiplayer Test Scaler uses two Amazon S3 buckets to store logs and metrics files generated by the server:
+
+1. A temporary "artifact" S3 bucket, defined by Multiplayer Test Scaler, where the contents of the `C:/o3de/user/log` and `C:/o3de/user/Metrics` directories on the server are synced every two minutes to ensure they are recoverable in case the server becomes unreachable during testing.
+1. A user-defined "export" bucket, external to Multiplayer Test Scaler, where the contents of the temporary artifact bucket are uploaded when testing is complete and the server stack is torn down. By default, an AWS CloudFormation stack output exported under the key **`O3deMetricsUploadBucket`** is looked up in the configured region and used as the upload destination. If such a bucket doens't exist in the region where you are deploying, you can either create it or specify a different export name for Multiplayer Test Scaler to look up:
```
Comment (lmbr-pip) [inline README.md:111]:
```suggestion
1. A user-defined "export" bucket, external to Multiplayer Test Scaler, where the contents of the temporary artifact bucket are uploaded when testing is complete and the server stack destroyed. By default, the code looks for an AWS CloudFormation stack output exported under the key **`O3deMetricsUploadBucket`** in the configured region, which is used as the upload destination. If such a bucket doesn't exist in the region where you are deploying, you can either create it or specify a different export name for Multiplayer Test Scaler to look up:
```

**Review by lmbr-pip — APPROVED** [review]

**Pull request merged**